GetUser/handler.ts

```diff
@@ -0,0 +1,301 @@
+              .listByService(
+                azureApimConfig.apimResourceGroup,
+                azureApimConfig.apim,
+                {
+                  filter: `email eq '${email}'`
+                }
+              )
+              .then(userList => ({
+                apiClient,
+                userList
+              })),
+          toError
+        )
+      )
+      .mapLeft<IResponseErrorInternal | IResponseErrorNotFound>(error =>
+        genericInternalErrorHandler(
+          context,
+          "GetUsers | Could not list the user by email.",
+          error,
+          genericErrorMessage
+        )
+      )
+      .chain(
+        fromPredicate(
+          taskResults => taskResults.userList.length !== 0,
+          () =>
+            ResponseErrorNotFound(
+              "Not found",
+              "The required resource does not exist"
+            )
+        )
+      )
+      .chain(taskResults =>
+        fromEither(NonEmptyString.decode(taskResults.userList[0].name))
+          .mapLeft(errors =>
+            genericInternalValidationErrorHandler(
+              context,
+              "GetUser | Could not get user name from user contract.",
+              errors,
+              genericErrorMessage
+            )
+          )
+          .map(userName => ({
+            apiClient: taskResults.apiClient,
+            userName
+          }))
+      )
+      .chain(taskResults =>
+        sequenceT(taskEither)(
+          getUserGroups(
+            taskResults.apiClient,
+            azureApimConfig.apimResourceGroup,
+            azureApimConfig.apim,
+            taskResults.userName
+          ),
+          getUserSubscriptions(
+            taskResults.apiClient,
+            azureApimConfig.apimResourceGroup,
+            azureApimConfig.apim,
+            taskResults.userName
+          )
+        ).mapLeft<IResponseErrorInternal | IResponseErrorNotFound>(error =>
+          genericInternalErrorHandler(
+            context,
+            "GetUsers | Could not get user groups and subscriptions from APIM.",
+            error,
+            genericErrorMessage
+          )
+        )
+      )
+      .map(contractLists => {
+        const [groupContracts, subscriptionContracts] = contractLists;
+        const errorOrGroups = array.traverse(either)(
+          [...groupContracts],
+          groupContractToApiGroup
+        );
+        const errorOrSubscriptions = array.traverse(either)(
+          [...subscriptionContracts],
+          subscriptionContractToApiSubscription
+        );
+        return { errorOrGroups, errorOrSubscriptions };
+      })
+      .chain(
+        fromPredicate(
+          userInfo => isRight(userInfo.errorOrGroups),
+          userInfoWithError =>
+            genericInternalErrorHandler(
+              context,
+              "GetUser | Invalid group contract from APIM.",
+              userInfoWithError.errorOrGroups.value as Error,
+              genericErrorMessage
```

Comment:
we should return the full error message to the api caller (Invalid group contract from APIM.), same for other calls of genericInternalErrorHandler.